src/ResourceStateBuilder.js

```diff
@@ -127,4 +173,46 @@ export default class ResourceStateBuilder {
     })
     return newDataset
   }
+
+  async findResourceTemplate(rtId) {
+    if (this.resourceTemplates[rtId]) {
+      return this.resourceTemplates[rtId]
+    }
+
+    return getResourceTemplate(rtId, 'ld4p').then((response) => {
+      const resourceTemplate = response.response.body
+      this.resourceTemplates[rtId] = resourceTemplate
+      // TODO: Validate template
```

Comment:
Do you want to take that on now or create a ticket for this?

Reply:
Ticketed.